pageserver/src/tenant/vectored_blob_io.rs

```diff
@@ -157,15 +163,28 @@ pub struct VectoredReadPlanner {
     // Arguments for previous blob passed into [`VectoredReadPlanner::handle`]
     prev: Option<(Key, Lsn, u64, BlobFlag)>,
 
-    max_read_size: usize,
+    max_read_size: Option<usize>,
 }
 
 impl VectoredReadPlanner {
     pub fn new(max_read_size: usize) -> Self {
         Self {
             blobs: BTreeMap::new(),
             prev: None,
-            max_read_size,
+            max_read_size: Some(max_read_size),
+        }
+    }
+
+    /// This function should *only* be used if the caller has a way to control the limit. e.g., in `ImageLayerIterator`,
```

Comment:
use references to ensure the comment doesn't bitrot

```suggestion
    /// This function should *only* be used if the caller has a way to control the limit. e.g., in [`crate::tenant::storage_layer::image_layer::ImageLayerIterator`],
```